internal/expr/parser.go

```diff
@@ -116,18 +116,53 @@ func (sa sliceAccessor) getTypeName() string {
 	return sa.typeName
 }
 
-// columnAccessor stores a SQL column name and optionally its table name.
-type columnAccessor struct {
-	tableName, columnName string
+type columnAccessor interface {
+	String() string
+	tableName() string
+	columnName() string
+}
+
+// staticColumn stores a SQL column name and optionally its table name.
+type staticColumn struct {
+	table, column string
+}
+
+func (sc staticColumn) columnName() string {
+	return sc.column
+}
+
+func (sc staticColumn) tableName() string {
+	return sc.table
 }
 
-func (ca columnAccessor) String() string {
-	if ca.tableName == "" {
-		return ca.columnName
+func (sc staticColumn) String() string {
+	if sc.table == "" {
+		return sc.column
 	}
-	return ca.tableName + "." + ca.columnName
+	return sc.table + "." + sc.column
+}
+
+var _ columnAccessor = staticColumn{}
+
+// sqlFunctionCall stores a function call that is used in place of a column.
+type sqlFunctionCall struct {
+	raw string
+}
+
+func (sfc sqlFunctionCall) columnName() string {
+	return sfc.raw
+}
+
+func (sfc sqlFunctionCall) tableName() string {
+	return ""
+}
+
+func (sfc sqlFunctionCall) String() string {
+	return sfc.raw
 }
 
+var _ columnAccessor = sqlFunctionCall{}
```

Comment:
At some point Joe advised that these should go in the tests rather than the main code, this can go in `parser_helper_test.go`

Reply:
Moved.